hedgehog/src/Hedgehog/Internal/Tree.hs

```diff
@@ -320,7 +339,7 @@ interleave :: forall m a. Monad m => [NodeT m a] -> NodeT m [a]
 interleave ts =
```

Comment:
`interleave` is begging for documentation. Detailed documentation would be great; barring that, a clear statement of its general goals would be helpful.